pkg/ipmi/ipmi.go

```diff
@@ -205,47 +204,102 @@ func fdSet(fd uintptr, p *syscall.FdSet) {
 	p.Bits[fd/64] |= 1 << (uint(fd) % 64)
 }
 
-func (i *IPMI) sendrecv(req *req) ([]byte, error) {
-	addr := systemInterfaceAddr{
+func ioctlSetReq(fd, name uintptr, req *req) error {
+	_, _, err := unix.Syscall(unix.SYS_IOCTL, fd, name, uintptr(unsafe.Pointer(req)))
+	runtime.KeepAlive(req)
+	if err != 0 {
+		return err
+	}
+	return nil
+}
+
+func ioctlGetRecv(fd, name uintptr, recv *recv) error {
+	_, _, err := unix.Syscall(unix.SYS_IOCTL, fd, name, uintptr(unsafe.Pointer(recv)))
+	runtime.KeepAlive(recv)
+	if err != 0 {
+		return err
+	}
+	return nil
+}
+
+// SendRecvBasic sends the IPMI message, receives the response, and returns the
+// response data. This is recommended for use unless the user must be able to
+// specify the data pointer and length on their own.
+func (i *IPMI) SendRecvBasic(netfn NetFn, cmd Command, data []byte) ([]byte, error) {
```

Comment:
Sorry, I know, churn, a lot of the naming for stuff like this usually ends up more like

SendRecvBasic -> SendRecv
SendRecv -> RawSendRecv

compare unix.Syscall vs unix.RawSyscall